helper/resource/testing.go

```diff
@@ -415,6 +418,14 @@ func LogOutput(t TestT) (logOutput io.Writer, err error) {
 // long, we require the verbose flag so users are able to see progress
 // output.
 func Test(t TestT, c TestCase) {
+	// If "TF_PARA" is set to some non-empty value, all the parallel test cases
+	// will be run parallel.
+	// example usage:
+	//	TF_PARA=1 go test [TEST] [TESTARGS] -parallel=n
+	if os.Getenv(ParaTestEnvVar) != "" {
```

Comment:
Hey @metacpp - this is where I think the logic could be changed if it was a flag in `TestCase`:

```
if c.Parallel {
  t.Parallel()
}
```

Reply:
@vancluever thanks for comments.

I was thinking about the possibility to allow each different acc tests to decide if it's parallel or not. For new test case to be added, this seems to be pretty good way. But for provider like azurerm, we have 618+ cases, all of them are qualified parallel cases. If we choose to enable the parallel run at test case level, we will need to change 618+ test cases in our code repo to set "Parallel" to be true. Even we use some text processing tool/script to do the work in a batch, @tombuildsstuff might have concern on the big change list and potential conflicts to be resolved if the test files are frequently changed, but we should be able to find out a way to resolve it.

I do have another suggestion here: instead of having "Parallel" as field, can we have "SkipParallel" instead ? The new code logic will be like:

if os.Getenv(ParaTestEnvVar) != "" && !SkipParallel {
  t.Parallel()
}

What do you think ?

Another solution is for each different provider, you can have a thin layer testing framework built on top of terra testing, which will have some default settings(ut, parallel) for its own scenario. But it will still need to change all the legacy test files in that provider repo, I am not sure if @tombuildsstuff  will agree so many changes.

Reply:
Hey @metacpp, will reply with most of the content out-line, but just wanted to mention that I don''t think

```
if os.Getenv(ParaTestEnvVar) != "" && !SkipParallel {
  t.Parallel()
}
```

Is a good approach too as this introduces 2 flags, rather than just one.

Will reply with the rest of the info in the main thread.

Reply:
@vancluever I did have a similar PR on azurerm provider, and I was suggested by @tombuildsstuff to do the PR in Terraform runtime coz we thought that this change might benefit more people who are using resource.Test in their repo. I am fine with introducing just one flag here, aka, IsParallel, just like IsUnitTest. And to save changes in each provider repo, we can discuss in below thread.